### PR TITLE
Fix spelling and grammar issues

### PR DESCRIPTION
**Pull request by p-e-w**
This PR fixes a bunch of spelling and grammar issues that I found with [LanguageTool](https://languagetool.org) using a semi-automated process that I will describe in detail below.

I have focused on issues that are obvious *mistakes,* rather than just poor style or against some convention. As such, this PR does **not** attempt to correct:

* British/American spelling inconsistencies, of which there are many.
* Violations of Oxford `ise`/`ize` word ending conventions.
* Missing commas, particularly in conjunctive clauses.
* Improper punctuation, such as hyphens being used in place of em-dashes.

The workflow described below finds hundreds of these issues, which may or may not be worth resolving as well. Note that I have thus far only looked at documents from the `specification` root directory, the `appendices` subdirectory, and the file `modules/end_to_end_encryption.rst`, as I unfortunately don't have more time to work on this for now.

---

### Workflow

Spell- and grammar-checking technical documents is challenging because symbols and jargon invariably generate an overwhelming number of false positives. The following setup works quite well for the Spec:

1. Install Pandoc and Java.
2. Download and unpack the standalone "Desktop" version of LanguageTool [from here](https://dev.languagetool.org/http-server).
3. Add [this custom Matrix.org Jargon Dictionary](https://gist.github.com/p-e-w/df3c4b2bb0f0aad242d1d014bb647463) to LanguageTool's bundled Hunspell version. The easiest way to accomplish this is by replacing the contents of `org/languagetool/resource/en/hunspell/ignore.txt` with the contents of that Gist. This dramatically reduces the number of false positives.
4. For each Spec document, run `pandoc -t plain PATH/TO/SPEC/DOCUMENT.rst | java -jar languagetool-commandline.jar -l en-GB -d 'WHITESPACE_RULE,DASH_RULE,EN_QUOTES,MULTIPLICATION_SIGN,WORD_CONTAINS_UNDERSCORE,ENGLISH_WORD_REPEAT_BEGINNING_RULE,COMMA_PARENTHESIS_WHITESPACE,SENTENCE_WHITESPACE,UPPERCASE_SENTENCE_START,PLUS_MINUS,ARROWS,OXFORD_SPELLING_ISE_VERBS,OXFORD_SPELLING_NOUNS,OXFORD_SPELLING_ADJECTIVES,LICENCE_LICENSE_NOUN_SINGULAR,AN_INVITE,UNLIKELY_OPENING_PUNCTUATION,EN_UNPAIRED_BRACKETS,ANYMORE'`.

The raised issues are of fairly high quality, with roughly 30-40% of them being actual problems that warrant fixing.

---

Signed-off-by: Philipp Emanuel Weidmann <pew@worldwidemann.com>

### CODE REVIEW

**Comment by p-e-w** [comment]
Do trivial changes like this require a news fragment? I've read CONTRIBUTING.rst but the answer is still not clear to me.

File: specification/appendices/identifier_grammar.rst [hunk 1]
```diff
@@ -379,7 +379,7 @@ are picked is left as an implementation detail, however the current recommendati
 to pick 3 unique servers based on the following criteria:
 
 * The first server should be the server of the highest power level user in the room,
-  provided they are at least power level 50. If no user meets this criteria, pick the
+  provided they are at least power level 50. If no user meets these criteria, pick the
```
Comment (KitsuneRal) [inline specification/appendices/identifier_grammar.rst:382]:
```suggestion
  provided they are at least power level 50. If no user meets this criterion, pick the
```

File: specification/proposals_intro.rst [hunk 2]
```diff
@@ -379,10 +379,10 @@ As part of the proposal process the spec core team will require evidence of the
 working in order for it to move into FCP. This can usually be a branch/pull request
 to whichever implementation of choice that proves the MSC works in practice, though
 in some cases the MSC itself will be small enough to be considered proven. Where it's
-unclear if a MSC will require an implementation proof, ask in `#matrix-spec:matrix.org
+unclear if an MSC will require an implementation proof, ask in `#matrix-spec:matrix.org
 <https://matrix.to/#/#matrix-spec:matrix.org>`_.
 
-Early release of a MSC/idea
+Early release of an MSC/idea
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (turt2live) [inline specification/proposals_intro.rst:386]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Review by turt2live — APPROVED** [review]
Thanks for doing this! I'll take care of the changelogs before merging.

**Pull request merged**